### ISSUE TITLE
Does Icarus support rand?

### ISSUE THREAD

**Issue by mickey4u** [steveicarus/iverilog#369]
I am using `rand` in my system verilog testbench and I am getting an error. Does Icarus support `rand`? if not what will be the alternative in achieving this?

**Comment by martinwhitaker** [comment]
No, it doesn't. You can declare random variables, but there is no support for the randomize() method. You will have to implement your own randomize() method, using the standard probabilistic distribution system functions (e.g. $random) to provide the random values.

**Comment by mickey4u** [comment]
@martinwhitaker Thanks for this. How do you declare random variables that Icarus supports.

**Comment by martinwhitaker** [comment]
As per the SV standard - prefix a class property declaration with the keyword rand. But of course, without support for randomize(), that is just decoration.

**Comment by mickey4u** [comment]
@martinwhitaker Okay I had that but Icarus flagged an error.
`    typedef struct packed {
        rand bit [15:0] a;
        bit [15:0] b;
    } cc;`

`
testbench.sv:42: syntax error
testbench.sv:42: Error in struct/union member.
`

**Comment by svenka3** [comment]
@mickey4u rand is a valid qualifier only inside class and not struct
